crates/sui-types/src/messages.rs

```diff
@@ -165,12 +168,27 @@ impl SingleTransactionKind {
                 .filter_map(|arg| match arg {
                     CallArg::Pure(_) => None,
                     CallArg::Object(ObjectArg::ImmOrOwnedObject(object_ref)) => {
-                        Some(InputObjectKind::ImmOrOwnedMoveObject(*object_ref))
+                        Some(vec![InputObjectKind::ImmOrOwnedMoveObject(*object_ref)])
                     }
                     CallArg::Object(ObjectArg::SharedObject(id)) => {
-                        Some(InputObjectKind::SharedMoveObject(*id))
+                        Some(vec![InputObjectKind::SharedMoveObject(*id)])
                     }
+                    CallArg::ObjVec(vec) => Some(
+                        vec.iter()
+                            .filter_map(|obj_arg| match obj_arg {
+                                ObjectArg::ImmOrOwnedObject(object_ref) => {
+                                    Some(InputObjectKind::ImmOrOwnedMoveObject(*object_ref))
+                                }
+                                ObjectArg::SharedObject(_) => {
+                                    // ObjVec is guaranteed to never contain shared objects
+                                    debug_assert!(false);
```

Comment:
Why do we have this restriction?

Reply:
I assumed that this PR is only meant to provide support for vectors of owned objects. I was not sure if there are any additional implications of supporting shared objects. On the surface there are not, but shared objects support is a bit newer (and a bit more complicated) so I did not want to make too many assumptions, so it would be good to get some advice on that.